port_ocean/clients/port/mixins/entities.py

```diff
@@ -58,9 +58,12 @@ async def upsert_entity(
             )
         handle_status_code(response, should_raise)
         result = response.json()
-        result_entity = Entity.parse_obj(result["entity"])
+        result_entity = Entity.parse_obj(result.get("entity", {}))
         # Set the results of the search relation and identifier to the entity
         entity.identifier = result_entity.identifier or entity.identifier
+        if isinstance(entity.identifier, dict):
```

Comment:
lets have a method inside of the entity class for knowing if its of type search identifier